### ISSUE TITLE
XML content is not trimmed

### ISSUE THREAD

**Issue by maixuanhan** [CESNET/libyang#308]
Please have option to trim the xml content when using `lyxml_parse_*` functions.

**Comment by rkrejci** [comment]
Can you provide some example? I don't know what exactly you want.

**Comment by maixuanhan** [comment]
Assume we have below hello message. And it is stored in `msg` variable.
```xml
   <hello xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
     <capabilities>
       <capability>
         urn:ietf:params:netconf:base:1.1
       </capability>
       <capability>
         urn:ietf:params:netconf:capability:startup:1.0
       </capability>
       <capability>
         http://example.net/router/2.3/myfeature
       </capability>
     </capabilities>
     <session-id>4</session-id>
   </hello>
```
Below code snippet is to retrieve the base capability. 
```c
struct lyxml_elem *elem = lyxml_parse_mem(ctx, msg, 0);

/* print the first <capability> element content */
printf("[%s]", elem->child->child->content);

/*
- It actually prints:
"[
         urn:ietf:params:netconf:base:1.1
       ]"
with pre and post unwanted white-space and line-feed characters

- I expect:
"[urn:ietf:params:netconf:base:1.1]"
*/
```

**Comment by rkrejci** [comment]
You cannot assume that white spaces in values has no meanings. Whatever prints the hello message, it prints it wrong. Fix the tool which prints this hello messages. Even `xmllint()` does not remove these whitespaces, try
```
$ xmllint --noblanks hello.xml
```

**Comment by rkrejci** [comment]
Just a note - however the trim functionality cannot be added into the libyang XML parser, the NETCONF server/client processing the hello messages with capabilities formatted this way are supposed to be processed and leading and trailing whitespaces are supposed to be removed. So we are now fixing this issue in libnetconf2 which processes hello messages provided by libyang XML parser. Anyway, we are not aware of any NETCONF implementation which would format hello messages this way.

**Comment by maixuanhan** [comment]
Above hello message is from RFC 6241 Network Configuration Protocol (NETCONF), RFC 6020 YANG - A Data Modeling Language for the Network Configuration Protocol (NETCONF). It should not be wrong.

I understand that you guys are keeping the xml parser compact and light but with trimming option, the parser will save our ass from extra string handling.

I am actually migrating my code from using libxml2 to libyang. With libxml2, I used `xmlReadMemory()` with `XML_PARSE_NOBLANKS ` option to parse the xml; I used `xmlNodeGetContent()` to get the content; I used `xmlStrEqual()` to compare... it's a bit annoying but the above hello is accepted.
